File: backend/src/main/java/br/com/hematsu/stock_flow/enums/TypeEnum.java
package br.com.hematsu.stock_flow.enums;

public enum TypeEnum {
    OUT(0),
    IN(1);

    private int code;

    private TypeEnum(int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    public static TypeEnum valueOf(int code) {
        for (TypeEnum value : TypeEnum.values()) {
            if (value.getCode() == code)
                return value;
        }
        throw new IllegalArgumentException("Tipo inválido");
    }

}

File: frontend/lib/models/enums/type_enum.dart
// ignore_for_file: constant_identifier_names

enum TypeEnum {
  OUT('Saída', 0),
  IN('Entrada', 1);

  final String displayName;
  final int code;

  const TypeEnum(this.displayName, this.code);

  static TypeEnum fromCode(int code) {
    return TypeEnum.values.firstWhere(
      (e) => e.code == code,
      orElse: () => throw ArgumentError('Código inválido: $code'),
    );
  }

  static TypeEnum fromString(String name) {
    return TypeEnum.values.firstWhere(
      (e) => e.name == name.toUpperCase(),
      orElse: () => throw ArgumentError('Tipo inválido: $name'),
    );
  }

  @override
  String toString() => name;
}
